cmd/tnf/generate/catalog/catalog.go

```diff
@@ -116,6 +121,85 @@ func emitTextFromFile(filename string) error {
 	return nil
 }
 
+// it turns a list of identifiers
+// { url, version  }
+// and takes urls like http://test-network-function.com/testcases/SuiteName/TestName
+// to build a more structured catalogue
+// {
+//     suiteNameA: [testName [identifiers], testName2 [identifiers] ]
+//     suiteNameB: [testName3 [identifiers], testName4 [identifiers] ]
+// }
+func createPrintableCatalogFromIdentifiers(keys []claim.Identifier) map[string][]catalogElement {
+	catalog := make(map[string][]catalogElement)
+	// we need the list of suite's names
+	var element catalogElement
+
+	for _, i := range keys {
+		suiteTest := identifiers.GetSuiteAndTestFromIdentifier(i)
+		if suiteTest == nil {
+			fmt.Fprintf(os.Stderr, "Identifier Url not valid\n")
+			return nil
+		}
+		suiteName := suiteTest[0]
+		testName := suiteTest[1]
+		element.testName = testName
+		element.identifier = i
+		catalog[suiteName] = append(catalog[suiteName], element)
+	}
+	return catalog
+}
+
+// it turns a list of urls http://test-network-function.com/tests/TestName
+// to build a more structured catalogue
+// {
+//     0: [testName [identifiers], testName2 [identifiers] ]
+//     1: [testName3 [identifiers], testName4 [identifiers] ]
+// }
+func createPrintableCatalogFromUrls(urls []string) map[int][]catalogElement {
+	catalog := make(map[int][]catalogElement)
+	var element catalogElement
+	var c = 0
+	for _, url := range urls {
+		testName := identifier.XformToGinkgoItIdentifier(identifier.Identifier{URL: url})
+		if testName == "" {
+			return nil
+		}
+		element.testName = testName
+		element.identifier = claim.Identifier{Url: url}
+		catalog[c] = append(catalog[c], element)
+		c++
```

Comment:
No need for `c` counter here, just use the index of the loop.

Reply:
done it